File: Codeforces/Email from Polycarp.py
n = int(input())
for _ in range(n):
    s = input() 
    t = input()
    s_keys = []
    s_values = []
    t_keys = []
    t_values = []
    i = 0
    while i < len(s): 
        same = 0
        j = i 
        while s[i] == s[j]:
            same += 1
            j += 1
            if j == len(s):
                break
        s_keys.append(s[i])
        s_values.append(same)
        i = j
        if i >= len(s):
            break
    i = 0
    while i < len(t): 
        same = 0
        j = i 
        while t[i] == t[j]:
            same += 1
            j += 1
            if j == len(t):
                break
        t_keys.append(t[i])
        t_values.append(same)
        i = j
        if i >= len(t):
            break
    if len(s_keys) != len(t_keys):
        print("NO")
        continue
    f = True
    """print(s_keys)
    print(s_values)
    print(t_keys)
    print(t_values)"""
    for i in range(len(s_keys)):
        if s_keys[i] != t_keys[i] or s_values[i] > t_values[i]:
            print("NO")
            f = False
            break
    if f:
        print("YES")

File: Codeforces/Email from Polycarp.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main(){
    int n;
    cin >> n;
    while (n--){
        string s, t;
        cin >> s >> t;
        vector<char>s_keys;
        vector<char>t_keys;
        vector<int>s_values;
        vector<int>t_values;
        int i, j, same, s_len, t_len;
        s_len = s.length();
        t_len = t.length();
        i = 0;
        while (i < s_len){
            //cout << i << "KOS" << endl;
            same = 0;
            j = i;
            while (s[i] == s[j]){
                same++;
                j++;
                if (j == s_len){
                    break;
                }
            }
            s_keys.push_back(s[i]);
            s_values.push_back(same);
            i = j;
            if (i >= s_len){
                break;
            } 
        }
        i = 0;
        while (i < t_len){
            //cout << i << "KOON" << endl;
            same = 0;
            j = i;
            while (t[i] == t[j]){
                same++;
                j++;
                if (j == t_len){
                    break;
                }
            }
            t_keys.push_back(t[i]);
            t_values.push_back(same);
            i = j;
            if (i >= t_len){
                break;
            }
        }
        if (s_keys.size() != t_keys.size()){
            cout << "NO" << endl;
            continue;
        }
        bool f = true;
        for (int i = 0; i<s_keys.size(); i++){
            if (s_keys[i] != t_keys[i] || s_values[i] > t_values[i]){
                cout << "NO" << endl;
                f = false;
                break;
            }
        }
        if (f){
            cout << "YES" << endl;
        }


    }
    return 0;
}